internal/cmd/signer.go

```diff
@@ -1,41 +1,37 @@
 package cmd
 
 import (
+	"context"
 	"errors"
 	"time"
 
 	"github.com/notaryproject/notation-go"
-	"github.com/notaryproject/notation-go/plugin/manager"
-	"github.com/notaryproject/notation-go/signature"
-	"github.com/notaryproject/notation/internal/envelope"
+	"github.com/notaryproject/notation-go/dir"
+	"github.com/notaryproject/notation-go/plugin"
+	"github.com/notaryproject/notation-go/signer"
 	"github.com/notaryproject/notation/pkg/configutil"
 )
 
 // GetSigner returns a signer according to the CLI context.
 func GetSigner(opts *SignerFlagOpts) (notation.Signer, error) {
-	// Construct a signer from key and cert file if provided as CLI arguments
-	mediaType, err := envelope.GetEnvelopeMediaType(opts.SignatureFormat)
-	if err != nil {
-		return nil, err
-	}
 	// Construct a signer from preconfigured key pair in config.json
 	// if key name is provided as the CLI argument
 	key, err := configutil.ResolveKey(opts.Key)
 	if err != nil {
 		return nil, err
 	}
 	if key.X509KeyPair != nil {
-		return signature.NewSignerFromFiles(key.X509KeyPair.KeyPath, key.X509KeyPair.CertificatePath, mediaType)
+		return signer.NewFromFiles(key.X509KeyPair.KeyPath, key.X509KeyPair.CertificatePath)
 	}
 	// Construct a plugin signer if key name provided as the CLI argument
 	// corresponds to an external key
 	if key.ExternalKey != nil {
-		mgr := manager.New()
-		runner, err := mgr.Runner(key.PluginName)
+		mgr := plugin.NewCLIManager(dir.PluginFS())
+		plugin, err := mgr.Get(context.Background(), key.PluginName)
```

Comment:
Do we need to check that is it a valid plugin by calling GetMetadata()?

Reply:
The workflow is: signer.NewFromPlugin -> pluginSigner.Sign -> plugin.GetMetadata. Therefore, it is validated inside notation-go. We should be fine here.